### PR TITLE
Do not emit deprecation warning in the shared context

### LINKED ISSUE #91383: [CI] SnapshotBasedRecoveryIT testSnapshotBasedRecovery failing
```
:qa:rolling-upgrade:v7.17.8#upgradedClusterTest FAILED	
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8	
REPRODUCE WITH: ./gradlew ':qa:rolling-upgrade:v7.17.8#upgradedClusterTest' -Dtests.class="org.elasticsearch.upgrades.SnapshotBasedRecoveryIT" -Dtests.method="testSnapshotBasedRecovery" -Dtests.seed=99DD5020A91464DB -Dtests.bwc=true -Dtests.locale=uk -Dtests.timezone=America/Curacao -Druntime.java=17	
org.elasticsearch.upgrades.SnapshotBasedRecoveryIT > testSnapshotBasedRecovery FAILED	
    org.elasticsearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:41245], URI [/snapshot_based_recovery/_settings], status line [HTTP/1.1 200 OK]	
    Warnings: [[index.indexing.slowlog.level] setting was deprecated in the previous Elasticsearch release and is removed in this release.]	
    {"acknowledged":true}	
        at __randomizedtesting.SeedInfo.seed([99DD5020A91464DB:703258D83588DF92]:0)	
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:343)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)	
        at app//org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1609)	
        at app//org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1603)	
        at app//org.elasticsearch.upgrades.SnapshotBasedRecoveryIT.testSnapshotBasedRecovery(SnapshotBasedRecoveryIT.java:103)	
Suite: Test class org.elasticsearch.upgrades.SnapshotBasedRecoveryIT	
  1> [2022-11-07T17:37:03,189][INFO ][o.e.u.SnapshotBasedRecoveryIT] [testSnapshotBasedRecovery] before test	
  1> [2022-11-07T17:37:03,193][INFO ][o.e.u.SnapshotBasedRecoveryIT] [testSnapshotBasedRecovery] initializing REST clients against [http://[::1]:36499, http://127.0.0.1:38953, http://[::1]:37855, http://127.0.0.1:43011, http://[::1]:45727, http://127.0.0.1:41245]	
  2> лист. 07, 2022 5:37:03 PM org.elasticsearch.client.RestClient logResponse	
  2> WARNING: request [PUT http://127.0.0.1:41245/snapshot_based_recovery/_settings] returned 1 warnings: [299 Elasticsearch-8.6.0-SNAPSHOT-a94347e002da18c9d567dde2b1bbec6b3ba133b0 "[index.indexing.slowlog.level] setting was deprecated in the previous Elasticsearch release and is removed in this release."]	
  1> [2022-11-07T17:37:03,387][INFO ][o.e.u.SnapshotBasedRecoveryIT] [testSnapshotBasedRecovery] There are still tasks running after this test that might break subsequent tests [health-node[c]].	
  1> [2022-11-07T17:37:03,388][INFO ][o.e.u.SnapshotBasedRecoveryIT] [testSnapshotBasedRecovery] after test	
  2> REPRODUCE WITH: ./gradlew ':qa:rolling-upgrade:v7.17.8#upgradedClusterTest' -Dtests.class="org.elasticsearch.upgrades.SnapshotBasedRecoveryIT" -Dtests.method="testSnapshotBasedRecovery" -Dtests.seed=99DD5020A91464DB -Dtests.bwc=true -Dtests.locale=uk -Dtests.timezone=America/Curacao -Druntime.java=17	
  2> org.elasticsearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:41245], URI [/snapshot_based_recovery/_settings], status line [HTTP/1.1 200 OK]	
    Warnings: [[index.indexing.slowlog.level] setting was deprecated in the previous Elasticsearch release and is removed in this release.]	
    {"acknowledged":true}	
        at __randomizedtesting.SeedInfo.seed([99DD5020A91464DB:703258D83588DF92]:0)	
        at app//org.elasticsearch.client.RestClient.convertResponse(RestClient.java:343)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)	
        at app//org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)	
        at app//org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1609)	
        at app//org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1603)	
        at app//org.elasticsearch.upgrades.SnapshotBasedRecoveryIT.testSnapshotBasedRecovery(SnapshotBasedRecoveryIT.java:103)	
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+main+multijob+platform-support-unix/os/debian-10&&immutable/qa/rolling-upgrade/build/testrun/v7.17.8#upgradedClusterTest/temp/org.elasticsearch.upgrades.SnapshotBasedRecoveryIT_99DD5020A91464DB-001	
  2> NOTE: test params are: codec=Asserting(Lucene94): {}, docValues:{}, maxPointsInLeafNode=1084, maxMBSortInHeap=7.319935879968938, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=uk, timezone=America/Curacao	
  2> NOTE: Linux 4.19.0-22-cloud-amd64 amd64/Oracle Corporation 17.0.2 (64-bit)/cpus=32,threads=1,free=361787920,total=536870912	
  2> NOTE: All tests run in this JVM: [DesiredNodesUpgradeIT, FeatureUpgradeIT, FieldCapsIT, IndexingIT, RecoveryIT, SnapshotBasedRecoveryIT]	
WARNING: A terminally deprecated method in java.lang.System has been called	
WARNING: System::setSecurityManager has been called by org.gradle.api.internal.tasks.testing.worker.TestWorker (file:/var/lib/jenkins/.gradle/wrapper/dists/gradle-7.5.1-all/1ehga6e77gqps5uk2kc5kf1vc/gradle-7.5.1/lib/plugins/gradle-testing-base-7.5.1.jar)	
WARNING: Please consider reporting this to the maintainers of org.gradle.api.internal.tasks.testing.worker.TestWorker	
WARNING: System::setSecurityManager will be removed in a future release	
Tests with failures:	
 - org.elasticsearch.upgrades.SnapshotBasedRecoveryIT.testSnapshotBasedRecovery
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/o3ol7dvdybvbu/tests/:qa:rolling-upgrade:v7.17.8%23upgradedClusterTest/org.elasticsearch.upgrades.SnapshotBasedRecoveryIT/testSnapshotBasedRecovery

**Reproduction line:**
`./gradlew ':qa:rolling-upgrade:v7.17.8#upgradedClusterTest' -Dtests.class="org.elasticsearch.upgrades.SnapshotBasedRecoveryIT" -Dtests.method="testSnapshotBasedRecovery" -Dtests.seed=99DD5020A91464DB -Dtests.bwc=true -Dtests.locale=uk -Dtests.timezone=America/Curacao -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.upgrades.SnapshotBasedRecoveryIT&tests.test=testSnapshotBasedRecovery

**Failure excerpt:**
```
org.elasticsearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:41245], URI [/snapshot_based_recovery/_settings], status line [HTTP/1.1 200 OK]
Warnings: [[index.indexing.slowlog.level] setting was deprecated in the previous Elasticsearch release and is removed in this release.]
{"acknowledged":true}

  at __randomizedtesting.SeedInfo.seed([99DD5020A91464DB:703258D83588DF92]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:343)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)
  at org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1609)
  at org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1603)
  at org.elasticsearch.upgrades.SnapshotBasedRecoveryIT.testSnapshotBasedRecovery(SnapshotBasedRecoveryIT.java:103)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### LINKED ISSUE #91636: [CI] RecoveryIT testRelocationWithConcurrentIndexing failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/g7zftqejxsph2/tests/:qa:rolling-upgrade:v7.17.8%23twoThirdsUpgradedTest/org.elasticsearch.upgrades.RecoveryIT/testRelocationWithConcurrentIndexing

**Reproduction line:**
```
./gradlew ':qa:rolling-upgrade:v7.17.8#twoThirdsUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.RecoveryIT" -Dtests.method="testRelocationWithConcurrentIndexing" -Dtests.seed=A6D532BC267FAB3C -Dtests.bwc=true -Dtests.locale=es-AR -Dtests.timezone=NZ-CHAT -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.upgrades.RecoveryIT&tests.test=testRelocationWithConcurrentIndexing

**Failure excerpt:**
```
org.elasticsearch.client.WarningFailureException: method [PUT], host [http://[::1]:45361], URI [/relocation_with_concurrent_indexing/_settings], status line [HTTP/1.1 200 OK]
Warnings: [[index.indexing.slowlog.level] setting was deprecated in the previous Elasticsearch release and is removed in this release.]
{"acknowledged":true}

  at __randomizedtesting.SeedInfo.seed([A6D532BC267FAB3C:1AB23BA34104C504]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:343)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)
  at org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1609)
  at org.elasticsearch.test.rest.ESRestTestCase.updateIndexSettings(ESRestTestCase.java:1603)
  at org.elasticsearch.upgrades.RecoveryIT.testRelocationWithConcurrentIndexing(RecoveryIT.java:239)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by idegtiarenko**
Corresponding fix in 7.17 branch: https://github.com/elastic/elasticsearch/pull/93190

Closes: #91383
Closes: #91636

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 1]
```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
...
-            "creating Index [{}], shards [{}]/[{}] - reason [{}]",
-            indexMetadata.getIndex(),
-            idxSettings.getNumberOfShards(),
-            idxSettings.getNumberOfReplicas(),
-            indexCreationContext
-        );
+        try (var ignored = threadPool.getThreadContext().stashContext()) {
+            final IndexSettings idxSettings = new IndexSettings(indexMetadata, settings, indexScopedSettings);
+            // we ignore private settings since they are not registered settings
+            indexScopedSettings.validate(indexMetadata.getSettings(), true, true, true);
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:704]: A deprecation header could be set here:
```
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:615)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:521)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:491)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndexService(IndicesService.java:703)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:622)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:175)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.createIndicesAndUpdateShards(IndicesClusterStateService.java:476)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:226)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:538)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:524)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:497)
```

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 2]
```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
...
-        // we ignore private settings since they are not registered settings
-        indexScopedSettings.validate(indexMetadata.getSettings(), true, true, true);
-        logger.debug(
-            "creating Index [{}], shards [{}]/[{}] - reason [{}]",
-            indexMetadata.getIndex(),
-            idxSettings.getNumberOfShards(),
-            idxSettings.getNumberOfReplicas(),
-            indexCreationContext
-        );
+        try (var ignored = threadPool.getThreadContext().stashContext()) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:701]: I think we want to stay in the same context, not start a fresh one:

```suggestion
        try (var ignored = threadPool.getThreadContext().newStoredContext()) {
```

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 3]
```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
...
-            indicesQueryCache,
-            mapperRegistry,
-            indicesFieldDataCache,
-            namedWriteableRegistry,
-            idFieldMappers.apply(idxSettings.getMode()),
-            valuesSourceRegistry,
-            indexFoldersDeletionListeners,
-            snapshotCommitSuppliers
-        );
     }
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:750]: Do we need to suppress warnings from this whole method, or could we reduce the scope?

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 4]
```diff
@@ -698,53 +698,55 @@ private synchronized IndexService createIndexService(
...
-            indexModule.addIndexEventListener(listener);
+            final IndexModule indexModule = new IndexModule(
+                idxSettings,
+                analysisRegistry,
+                getEngineFactory(idxSettings),
+                directoryFactories,
+                () -> allowExpensiveQueries,
+                indexNameExpressionResolver,
+                recoveryStateFactories
+            );
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:721]: As well as here:
```
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.index.IndexModule.IndexModule.<init>>(IndicesService.java:713)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndexService(IndicesService.java:713)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:622)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.IndicesService.createIndex(IndicesService.java:175)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.createIndicesAndUpdateShards(IndicesClusterStateService.java:476)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:226)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:538)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:524)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:497)
```

**Review by DaveCTurner — COMMENTED** [review]
We assert that the task executor doesn't leak response headers here:

https://github.com/elastic/elasticsearch/blob/73863dda7be0a03d89150b1a66cb9407c7ab55d7/server/src/main/java/org/elasticsearch/cluster/service/MasterService.java#L1080-L1087

I think this bug indicates we need something similar here:

https://github.com/elastic/elasticsearch/blob/454b3e610a13c809add90712a6a1f60a6e2b5377/server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java#L537-L539

and here:

https://github.com/elastic/elasticsearch/blob/454b3e610a13c809add90712a6a1f60a6e2b5377/server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java#L557-L559

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java (hunk 4 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java:721]: In both cases deprecation headers remain in thread context until `callClusterStateListeners` is executed that might contain listeners for unrelated changes (such as other index create or update operations).

Similar issue exists in 7.17 branch: https://github.com/elastic/elasticsearch/pull/93190

Warning should be whitelisted for all 7.17 versions that are already released.

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java (hunk 3 above)
Comment (idegtiarenko) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java:750]: Highlighted the places that emit the warning. I guess, it could be reduced

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java:754]: Thanks for the extra detail. That's pretty much the whole method, let's leave it as it is.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java:754]: Oh, wait, I see, this is a new method that just creates the `IndexModule`. Great, thanks for isolating it like this.

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 5]
```diff
@@ -740,13 +721,38 @@ private synchronized IndexService createIndexService(
...
-            idFieldMappers.apply(idxSettings.getMode()),
+            idFieldMappers.apply(indexModule.indexSettings().getMode()),
             valuesSourceRegistry,
             indexFoldersDeletionListeners,
             snapshotCommitSuppliers
         );
     }
 
+    private IndexModule createIndexModule(IndexService.IndexCreationContext indexCreationContext, IndexMetadata indexMetadata) {
+        try (var ignored = threadPool.getThreadContext().newStoredContext()) {
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:732]: Hmm, although we do use this within a cluster state applier, I just remembered that we also use it to validate things before creating an index. It's important that deprecation warnings from this validation process do propagate back to the caller.

**Comment by idegtiarenko** [comment]
@DaveCTurner , regarding assertions: I am getting a lot of test failures. Is it okay to split that into separate issue and pr?

**Comment by DaveCTurner** [comment]
Yes, sure

File: qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/UpgradeWithOldIndexSettingsIT.java [hunk 6]
```diff
@@ -111,4 +112,20 @@ private void assertCount(String index, int countAtLeast) throws IOException {
 
         assertTrue(hitsTotal >= countAtLeast);
     }
+
+    public static void updateIndexSettingsWithPossibleDeprecation(String index, Settings.Builder settings) throws IOException {
```
Comment (DaveCTurner) [inline qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/UpgradeWithOldIndexSettingsIT.java:116]: Naming nit: maybe `updateIndexSettingsPermittingSlowlogDeprecationWarning`?

**Review by DaveCTurner — APPROVED** [review]
I would have preferred a way to make sure no other update-settings calls fall into this trap, maybe including ones added in future. I think would be possible by, for instance, wrapping up the `RestClient` used in the REST tests. However, there is no `FilterRestClient` today, and it seems like a lot of effort to introduce one just for this purpose, so let's go with this for now. We can revisit this question in future if it turns out to be a problem.

**Pull request merged**